humility-cmd/src/hiffy.rs

```diff
@@ -399,6 +399,89 @@ impl<'a> HiffyContext<'a> {
         Ok(())
     }
 
+    /// Convenience routine to translate an Idol call into HIF operations
+    pub fn idol_call_ops_read(
+        &self,
+        funcs: &HiffyFunctions,
+        op: &idol::IdolOperation,
+        payload: &[u8],
+        ops: &mut Vec<Op>,
+        lease_size: u32,
+    ) -> Result<()> {
+        self.idol_call_ops_rw(
+            funcs,
+            op,
+            payload,
+            ops,
+            lease_size,
+            "SendLeaseRead",
+        )
+    }
+
+    pub fn idol_call_ops_write(
+        &self,
+        funcs: &HiffyFunctions,
+        op: &idol::IdolOperation,
+        payload: &[u8],
+        ops: &mut Vec<Op>,
+        lease_size: u32,
+    ) -> Result<()> {
+        self.idol_call_ops_rw(
+            funcs,
+            op,
+            payload,
+            ops,
+            lease_size,
+            "SendLeaseWrite",
+        )
+    }
+
+    /// Convenience routine to translate an Idol call into HIF operations
+    pub fn idol_call_ops_rw(
+        &self,
+        funcs: &HiffyFunctions,
+        op: &idol::IdolOperation,
+        payload: &[u8],
+        ops: &mut Vec<Op>,
+        lease_size: u32,
```

Comment:
Splitting off the no-lease/read/write paths this low in the stack appears to create a lot of new parallel APIs upwards.  I suspect this PR would be a lot smaller if you:
* added a `lease_size: Option<u32>` argument to idol_call_ops()
* added `read_size: Option<usize` and `write_data: Option<&[u8]>` to hiffy_call()
* changed hiffy_call() return type to `Result<std::result::Result<(humility::reflect::Value, Option<Vec<u8>>), String>>`
* use an enum for hiffy_call()'s Result error type to indicate lease mismatches to the caller

Reply:
Good suggestion – I didn't do _exactly_ this, but consolidated a bunch in 836c1f65710561ed604f62e411140404b6be4dc9 and c78ff03235f46e9592b04ad73114f99de106de4f